cli/lsp/language_server.rs

```diff
@@ -224,6 +232,9 @@ impl LanguageServer {
     referrer: ModuleSpecifier,
     force_global_cache: bool,
   ) -> LspResult<Option<Value>> {
+    if !self.init_token.is_cancelled() {
+      self.init_token.cancelled().await;
+    }
```

Comment:
It's a little strange reading this code because it's talking about cancellation. Maybe we should create some primitive somewhere called `AsyncFlag` or something that wraps the usage of `CancellationToken` then change this line to something like?

```rs
self.init_token.wait_raised().await;
```

Then to raise it we can do `self.init_token.raise()`?